components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,462 @@
+-- @Liquipedia
+-- wiki=trackmania
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local CustomMatchGroupInput = {}
+
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Array = require('Module:Array')
+local Lua = require('Module:Lua')
+local PageVariableNamespace = require('Module:PageVariableNamespace')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local TypeUtil = require('Module:TypeUtil')
+local Variables = require('Module:Variables')
+local Streams = require('Module:Links/Stream')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+
+local STATUS_HAS_SCORE = 'S'
+local STATUS_DEFAULT_WIN = 'W'
+local ALLOWED_STATUSES = { STATUS_DEFAULT_WIN, 'FF', 'DQ', 'L' }
+local STATUS_TO_WALKOVER = { FF = 'ff', DQ = 'dq', L = 'l' }
+local MAX_NUM_OPPONENTS = 2
+local MAX_NUM_PLAYERS = 10
+local MAX_NUM_VODGAMES = 20
+local RESULT_TYPE_DRAW = 'draw'
+
+local globalVars = PageVariableNamespace()
+
+-- containers for process helper functions
+local matchFunctions = {}
+local mapFunctions = {}
+local opponentFunctions = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(match)
+	Table.mergeInto(
+		match,
+		matchFunctions.readDate(match)
+	)
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+	match = matchFunctions.getExtraData(match)
+
+	return match
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processMap(map)
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+	map = mapFunctions.getTournamentVars(map)
+	map = mapFunctions.getParticipantsData(map)
+
+	return map
+end
+
+function CustomMatchGroupInput.processOpponent(record, date)
+	local opponent = Opponent.readOpponentArgs(record)
+		or Opponent.blank()
+
+	-- Convert byes to literals
+	if opponent.type == Opponent.team and opponent.template:lower() == 'bye' then
+		opponent = {type = Opponent.literal, name = 'BYE'}
+	end
+
+	Opponent.resolve(opponent, date)
+	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+
+	--score2 & score3 support for every match
+	local score2 = tonumber(record.score2 or '')
+	local score3 = tonumber(record.score3 or '')
+	if score2 then
+		record.extradata = {
+			score2 = score2,
+			score3 = score3,
+			set1win = Logic.readBool(record.set1win),
+			set2win = Logic.readBool(record.set2win),
+			set3win = Logic.readBool(record.set3win),
+			additionalScores = true
+		}
+	end
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processPlayer(player)
+	return player
+end
+
+--
+--
+-- function to sort out winner/placements
+function CustomMatchGroupInput._placementSortFunction(table, key1, key2)
```

Comment:
As it is a sorting function, isn't it a good usage ? Otherwise it is only used one time, so var name should be something like `opponents`, `opkey1` and `opkey2`

Reply:
yeah i was thinking something like 
`table` --> `opponents`
and `key1` --> `opponent1Key`
and `key2` --> `opponent2Key`

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/1d9003cc7e61a67cf067358bee20721daf02192f